R/layers.R

```diff
@@ -177,24 +177,125 @@ popupOptions = function(
   )
 }
 
-#' @param icon the icon for markers; if you want to create a new icon using
-#'   JavaScript, please remember to use \code{\link[htmlwidgets]{JS}()} on the
-#'   JavaScript string; see \url{http://leafletjs.com/reference.html#icon}
+#' @param icon the icon(s) for markers; you can either use literal JavaScript
+#'   code (e.g. \code{\link[htmlwidgets]{JS}('L.icon({iconUrl: '?', size: [?,
+#'   ?]})')} or an R list to create an icon (e.g. \code{list(iconUrl = '?', size
+#'   = c(x, y))}), or use \code{\link{iconList}()} to create a list of icons;
```

Comment:
Oh, looking at `iconList()` I think you'd still need it, but it would return a named list.